packages/ui5-application-writer/src/options.ts

```diff
@@ -5,13 +5,13 @@ import type { Ui5App } from './types';
 import { getFilePaths } from './files';
 import type { UI5Config } from '@sap-ux/ui5-config';
 import { ui5NPMSupport, ui5TSSupport } from './data/ui5Libs';
-import { mergeObjects, UI5_DEFAULT } from './data/defaults';
+import { getEsmTypesVersion, mergeObjects, UI5_DEFAULT } from './data/defaults';
 
 /**
  * Input required to enable optional features.
  */
 export interface FeatureInput {
-    ui5App: { app: { id: string; baseComponent?: string } };
+    ui5App: { app: { id: string; baseComponent?: string }; ui5?: { esmTypesVersion?: string; minUI5Version?: string } };
```

Comment:
Using the FeatureInput to handle the change locally. The input object is being shared across the different factories so a new variable introduced to handle this particular flow for the specific TS template.